Api/RequestHandler.php

```diff
@@ -407,8 +416,35 @@ protected function formatTotal($amount)
         if (empty($amount)) {
             return;
         }
+        
+        $value = $this->currency->formatTxt($amount, ['display' => \Magento\Framework\Currency::NO_SYMBOL]);
 
-        return $this->currency->formatTxt($amount, ['display' => \Magento\Framework\Currency::NO_SYMBOL]);
+        $separatorComa = strpos($value, ',');
+        $separatorDot  = strpos($value, '.');
+
+        if ($separatorComa !== false && $separatorDot !== false) {
+            if ($separatorComa > $separatorDot) {
+                $price = preg_replace("/(\d+)\.(\d+),(\d+)/", "$1,$2.$3", $value);
+            } else {
+                $price = $value;
+            }
+        } elseif ($separatorComa !== false) {
+            $locale = $this->_localeResolver->getLocale();
+
+            /**
+             * It's hard code for Japan locale.
+             */
+            $price = number_format(
+                (float)
+                str_replace(',', $locale === self::JAPAN_LOCALE_CODE ? '' : '.', $value),
+                2,
+                '.',
+                ','
+            );
+        } else {
+            $price = str_replace(",", ".", $value);
```

Comment:
@samumaretiya Is this condition needed? Seems that it would only execute when `$separatorDot !== false`, but in that case the line does nothing since there is no comma to replace

Reply:
Hi @soleilcot , I've updated the code. Could you please review. thanks, sam